third_party/terraform/resources/resource_google_billing_subaccount.go

```diff
@@ -0,0 +1,147 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"google.golang.org/api/cloudbilling/v1"
+)
+
+func resourceBillingSubaccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBillingSubaccountCreate,
+		Read:   resourceBillingSubaccountRead,
+		Delete: resourceBillingSubaccountDelete,
+		Update: resourceBillingSubaccountUpdate,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"master_billing_account": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"deletion_policy": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "",
+				ValidateFunc: validation.StringInSlice([]string{"RENAME_ON_DESTROY", ""}, false),
+			},
+			"billing_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"open": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceBillingSubaccountCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	displayName := d.Get("display_name").(string)
+	masterBillingAccount := d.Get("master_billing_account").(string)
+
+	billingAccount := &cloudbilling.BillingAccount{
+		DisplayName:          displayName,
+		MasterBillingAccount: canonicalBillingAccountName(masterBillingAccount),
+	}
+
+	res, err := config.NewBillingClient(userAgent).BillingAccounts.Create(billingAccount).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating billing subaccount '%s' in master account '%s': %s", displayName, masterBillingAccount, err)
+	}
+
+	d.SetId(res.Name)
+
+	return resourceBillingSubaccountRead(d, meta)
+}
+
+func resourceBillingSubaccountRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	id := d.Id()
+
+	billingAccount, err := config.NewBillingClient(userAgent).BillingAccounts.Get(d.Id()).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Billing Subaccount Not Found : %s", id))
+	}
+
+	d.Set("name", billingAccount.Name)
```

Comment:
Mind checking the error returned from these `d.Set` calls, and returning a message including the field name if there's a problem? Like in https://github.com/hashicorp/terraform-provider-google-beta/blob/master/google-beta/resource_container_cluster.go#L1584.

Reply:
added error checking for `d.Set` calls